aten/src/ATen/native/Blas.cpp

```diff
@@ -129,4 +138,22 @@ Tensor dot(const Tensor &self, const Tensor &other){
   });
 }
 
+Tensor vdot(const Tensor &self, const Tensor &other){
+  at::NoNamesGuard guard;
+
+  // Dispatch to `dot` for real dtypes.
+  if (!self.is_complex()){
+    return at::dot(self, other);
+  }
+
+  // For complex dtypes.
+  dot_check(self, other);
+  return AT_DISPATCH_COMPLEX_TYPES(self.scalar_type(), "vdot", [&] {
+    Tensor result = at::empty({}, self.options());
```

Comment:
`at::empty(0, ...`

This actually trips up experienced PyTorch developers all the time. If we take a look at `empty`:

https://github.com/pytorch/pytorch/blob/eb47940c0a773ec52e67778dbe01918f6b162271/aten/src/ATen/native/TensorFactories.cpp#L102

the size comes from here:

https://github.com/pytorch/pytorch/blob/eb47940c0a773ec52e67778dbe01918f6b162271/aten/src/ATen/native/TensorFactories.cpp#L121

which is computed using

https://github.com/pytorch/pytorch/blob/c9a1fc2b31ded472c63b227ad24859bc14131f42/aten/src/ATen/Utils.h#L96

So if we run the following program:

```
std::cout << prod_intlist({}) << std::endl;
  std::cout << prod_intlist(0) << std::endl;
```

it will print 1, 0. It is kinda confusing that an empty initializer list produces a tensor with one element and putting a zero in produces a tensor with no elements, but I try to remember it by recalling that zero specifies the size.

Reply:
Oh wait. You're immediately populating the value into result, not resizing it (which I assumed since it's such a common pattern). You're doing everything right.

Reply:
(Although what about `empty(1, ...` for readability?)